example/src/main/java/io/netty/example/http/helloworld/HttpHelloWorldServerHandler.java

```diff
@@ -52,11 +50,12 @@ public void channelRead0(ChannelHandlerContext ctx, HttpObject msg) {
             response.headers().set(CONTENT_TYPE, "text/plain");
             response.headers().setInt(CONTENT_LENGTH, response.content().readableBytes());
 
-            if (!keepAlive) {
-                ctx.write(response).addListener(ChannelFutureListener.CLOSE);
-            } else {
-                response.headers().set(CONNECTION, KEEP_ALIVE);
+            if (keepAlive) {
                 ctx.write(response);
+            } else {
+                // Tell the client we're going to close the connection.
+                response.headers().set(CONNECTION, CLOSE);
+                ctx.write(response).addListener(ChannelFutureListener.CLOSE);
```

Comment:
@normanmaurer Another question here, is it generally safe to close the connection after "write" operation instead of after "flush"?

Reply:
yes as the write is only be "executed" when the flush happens. Executed here means that it is "written to the socket".

Reply:
Okay, cool! I'm checking just because this is the only place we're adding the listener to `.write` when in other cases to `.writeAndFlush`.